src/ocamldep.ml

```diff
@@ -164,15 +202,24 @@ let deps_of cctx ~ml_kind unit =
         );
       let build_paths dependencies =
         let dependency_file_path m =
-          let path =
+          let file_path m =
             if is_alias_module cctx m then
               None
             else
               match Module.file m Ml_kind.Intf with
               | Some _ as x -> x
-              | None -> Module.file m Ml_kind.Impl
+              | None ->
+                Module.file m Ml_kind.Impl
           in
-          Option.map path ~f:all_deps_path
+          let module_file_ =
+            match file_path m with
+            | Some v -> Some v
+            | None ->
```

Comment:
When can this happen?

Reply:
This should happen quite regularly. It's the case when we our implementation depends on some module that is in the virtual library. This module doesn't have a source, but it still has object files.

Reply:
I don't follow the rest of the code though: we get here if we are generating the rules for an implementation and `m` is a module that is implemented in the virtual library. In this case, it is definitely not going to be part of `Compilation_context.virtual_modules cctx`, some this branch will always return `None`

Reply:
I think the name is just bad. This is just the `modules` field for the virtual library. So `virtual_modules` doesn't include just the virtual modules of the virtual library.

Reply:
I chose the name `vlib_modules`. I think this is still clear while being a lot shorter.